tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/types/constant/ConstantAssignmentTest.java

```diff
@@ -103,11 +103,30 @@ public void testConstantAssignmentViaConstant() {
 
     @Test
     public void testConstantAssignmentNegative() {
-        BAssertUtil.validateError(negativeCompileResult, 0, "incompatible types: expected 'int', found 'float'", 1, 16);
-        BAssertUtil.validateError(negativeCompileResult, 1, "incompatible types: expected 'float', found 'string'", 3,
-                31);
-        BAssertUtil.validateError(negativeCompileResult, 2, "incompatible types: expected 'int', found 'string'", 5,
-                27);
+        int i = 0;
+        BAssertUtil.validateError(negativeCompileResult, i++, "incompatible types: expected 'int', found 'float'",
+                1, 16);
+        BAssertUtil.validateError(negativeCompileResult, i++, "incompatible types: expected 'float', found 'string'",
+                3, 31);
+        BAssertUtil.validateError(negativeCompileResult, i++, "incompatible types: expected 'int', found 'string'",
+                5, 27);
+//        BAssertUtil.validateError(negativeCompileResult, i++, "missing non-defaultable required record field 'a'",
```

Comment:
Remove if not used, otherwise, comment with the issue link.

Reply:
We have restricted the `record-type` as expected type due to the default-value issue. I will create an issue and mention it.

Reply:
Removed [dbb563c](https://github.com/ballerina-platform/ballerina-lang/pull/39589/commits/dbb563c5fd05a206f691be51f3a3fdb07b082e18)